src/Adapter/Shop/QueryHandler/SearchShopsHandler.php

```diff
@@ -59,17 +68,28 @@ public function __construct(ShopRepository $shopRepository)
     public function handle(SearchShops $query): array
     {
         $searchTerm = $query->getSearchTerm();
+        $shopGroupList = $this->shopGroupRepository->findBySearchTerm($searchTerm);
         $shopList = $this->shopRepository->findBySearchTerm($searchTerm);
         $result = [];
 
+        foreach ($shopGroupList as $shopGroup) {
+            if (!$shopGroup->getShops()->isEmpty()) {
+                $result[] = new FoundShopGroup(
+                    $shopGroup->getId(),
+                    $shopGroup->getColor() ?? '',
+                    $shopGroup->getName()
+                );
+            }
+        }
+
         foreach ($shopList as $shop) {
             $result[] = new FoundShop(
-                $shop['id'],
-                $shop['color'] ?? '',
-                $shop['name'],
-                $shop['shopGroup']['id'],
-                $shop['shopGroup']['name'],
-                $shop['shopGroup']['color']
+                $shop->getId(),
+                $shop->getColor() ?? '',
+                $shop->getName(),
+                $shop->getShopGroup()->getId(),
+                $shop->getShopGroup()->getName(),
+                $shop->getShopGroup()->getColor()
```

Comment:
Why not embed a `FoundShopGroup` in `FoundShop` ?

Reply:
it would not be very practical, this is used by the autocomplete widget, which renders both `ShopGroup` and `Shop` distinctly.

Reply:
OK ;)